www/src/components/__tests__/docs-markdown-page.js

```diff
@@ -0,0 +1,161 @@
+import React from "react"
+import { render } from "@testing-library/react"
+import { Helmet } from "react-helmet"
+import DocsMarkdownPage from "../docs-markdown-page"
+import { ThemeProvider } from "theme-ui"
+import { I18nProvider } from "@lingui/react"
+import { MDXProvider } from "@mdx-js/react"
+
+import theme from "../../../src/gatsby-plugin-theme-ui"
+
+jest.mock("gatsby-plugin-mdx", () => ({
+  MDXRenderer: () => <div />
+}));
+
+jest.mock("../../utils/sidebar/item-list", () => {
+  return {
+    itemListContributing: {},
+    itemListDocs: {},
+    getItemList: jest.fn(),
+  }
+})
+
+jest.mock("../../hooks/use-site-metadata", () => {
+  return () => {
+    return { siteUrl: "https://www.gatsbyjs.org" }
+  }
+})
+
+Object.defineProperty(window, "IntersectionObserver", {
```

Comment:
Hmm, I see that this is copied from the table of contents tests. Ideally we'd do something like what is recommended [in the Jest docs](https://jestjs.io/docs/en/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom) but we can leave that for a separate PR.

Reply:
Yes! I agree to create another PR for that and maybe we can also add a test-utils file or similar as shown in react-testing-library [docs](https://testing-library.com/docs/react-testing-library/setup) where we could define a custom render with all the needed providers since now is repeated in the tests. What do you think @tesseralis ?